cmds/pxeboot/pxeboot.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/u-root/u-root/pkg/dhclient"
+	"github.com/u-root/u-root/pkg/kexec"
+	"github.com/vishvananda/netlink"
+)
+
+func kexecReader(kernel io.ReadCloser, cmdline string) error {
+	// Have to pass an FD to kexec file load.
+	r, w, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+
+	done := make(chan error)
+	go func() {
+		defer close(done)
+		defer kernel.Close()
+		_, err := io.Copy(w, kernel)
+		done <- err
+	}()
+
+	err = kexec.FileLoad(r, nil, cmdline)
+	cErr := <-done
+	if cErr != nil {
+		// We assume a copy error leads to a kexec load error. If it
+		// doesn't, we've got bigger problems.
+		return fmt.Errorf("error copying: %v | error kexec loading: %v", cErr, err)
+	}
+	if err != nil {
+		return err
+	}
+	return kexec.Reboot()
+}
+
+func Netboot() error {
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		return err
+	}
+
+	for _, name := range ifnames {
+		// DHCP lease
+		client, err := dhclient.New(dhclient.V4, name, 30*time.Second, 5)
```

Comment:
If you specify V4 here, don't you get a client who's solicit will use the V4 solicit? and packet.Boot right now doesn't support V4 right?